src/patterns/InPlaceFilters/FilterButton.tsx

```diff
@@ -0,0 +1,204 @@
+import * as Popover from '@radix-ui/react-popover';
+import classnames from 'classnames';
+import React from 'react';
+import Button from '../../components/Button/Button';
+import { ButtonVariants } from '../../components/Button/types';
+import Drawer from '../../components/Drawer/Drawer';
+import Filter from '../../components/Filter/Filter';
+import FilterHeader from '../../components/Filter/FilterHeader';
+import FilterInput from '../../components/Filter/FilterInput';
+import { SSRMediaQuery } from '../../providers/SeldonProvider/utils';
+import { px } from '../../utils';
+import FilterMenu from '../FilterMenu/FilterMenu';
+import { FilterButtonDisplay } from './FilterButtonDisplay';
+import { FilterDropdown } from './FilterDropdown';
+import { FilterButtonProps, FilterDimension, FilterType } from './types';
+import { countActiveFilters } from './utils';
+
+export const FilterButton = React.forwardRef<HTMLButtonElement, FilterButtonProps>(
+  (
+    {
+      id,
+      className,
+      filterButtonLabel,
+      buttonType,
+      handleClick,
+      filtersListState,
+      index,
+      filters,
+      handleFilterSelection,
+      handleFilterUpdate,
+      clearFilterUpdate,
+      resultsCount,
+      ariaLabels = {},
+    },
+    ref,
+  ) => {
+    const isButtonSelected = filtersListState?.[index ?? -1] ?? false;
+
+    const handleFilterButtonClick = () => {
+      if (filtersListState && handleClick) {
+        handleClick(filtersListState.map((selected, i) => (i === index ? !selected : false)));
+      }
+    };
+
+    const { totalCount, filterCount } = countActiveFilters(filters, buttonType);
+    const buttonLabel =
+      `${filterButtonLabel === 'Sort' ? 'Sort By' : filterButtonLabel}` + (filterCount > 0 ? ` (${filterCount})` : '');
+
+    // Main Filter button (shows all filters in a Drawer)
+    if (buttonType === 'Filter' && filters) {
+      return (
+        <>
+          <FilterButtonDisplay
+            ref={ref}
+            className={className}
+            isButtonSelected={isButtonSelected}
+            filterCount={filterCount}
+            totalCount={totalCount}
+            buttonType={buttonType}
+            filterButtonLabel={filterButtonLabel}
```

Comment:
couldn't these just be `label` here and `onClick`, `isSelected` for the props that the `<FilterButtonDisplay`